earthmover/nodes/node.py

```diff
@@ -114,12 +120,27 @@ def post_execute(self, **kwargs):
         else:
             self.num_rows, self.num_cols = self.data.shape
 
+        # Only actually compute() and count the rows if `require_rows` was defined for this node.
+        if self.require_rows > 0:
+            self.num_rows = dask.compute(self.num_rows)[0]
```

Comment:
Could we put line 125 into `self.check_require_rows()`?

Reply:
Sure, done.